src/FSDir.cpp

```diff
@@ -0,0 +1,33 @@
+#include "FSDir.h"
+#include "FSFile.h"
+
+FSDir::FSDir(std::string path):FSItem(path){
+	this->finfo.get()->st.get()->st_mode = S_IFDIR | 0755;
+	this->finfo.get()->st.get()->st_nlink = 1;
+	this->finfo.get()->st.get()->st_size=0;
+	this->finfo->st->st_atime = time( NULL );
+	this->finfo->st->st_mtime = time( NULL );
+	
+}
+void FSDir::createFile(std::string fname){
+	std::shared_ptr<FSFile> f = std::make_shared<FSFile>(fname);
+	this->flist.insert(std::pair<std::string,std::shared_ptr<FSFile> >(fname,f));
```

Comment:
Consider using emplace instead of insert to avoid creating the std::pair object explicitly